codegen/templates/chart/operator-deployment.yamltmpl

```diff
@@ -237,6 +249,36 @@ spec:
 {{ include "skv2.utils.merge" (list . $[[ $operatorVar ]]ServiceOverrides "[[ $operator.Name ]].serviceSpec") }}
 {{- end }}
 
----
-
-[[ end ]]
\ No newline at end of file
+[[- range $container := $containers -]]
+[[- if and (gt (len $container.EnableStatement) 0) (gt (len $operator.Service.Ports) 0) ]]
+[[ printf "\n---" ]]
+[[ $containerVar := printf "$.Values.%s" (lower_camel $container.Name) ]]
+[[- printf "\n{{- if %s -}}" $container.EnableStatement ]]
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: [[ $container.Name ]]
+    [[- range $key, $value := $container.Service.CustomLabels ]]
+    [[ $key ]]: [[ $value ]]
+    [[- end ]]
+  annotations:
+    app.kubernetes.io/name: [[ $container.Name ]]
+    [[- range $key, $value := $container.Service.CustomAnnotations ]]
+    [[ $key ]]: [[ $value ]]
+    [[- end ]]
+  name: [[ $container.Name ]]
+  namespace: {{ default .Release.Namespace $[[ $operatorVar ]].namespace }}
+spec:
+  selector:
+    app: [[ $operator.Name ]]
+  type: {{ [[ $containerVar ]].serviceType }}
+  ports:
+  [[- range $port := $container.Service.Ports ]]
+  - name: [[ lower $port.Name ]]
+    port: {{ [[ $containerVar ]].ports.[[ $port.Name ]] }}
+  [[- end ]]
+{{- end }}
+[[- end ]]
+[[- end ]]
+[[ end ]]
```

Comment:
This seems like a redundant Service, your test doesn't seem to reference the container.Service anyways so I am not sure what this is for, was this intended to be the sidecar?